source/material_model/modified_tait.cc

```diff
@@ -0,0 +1,282 @@
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+#include <aspect/material_model/modified_tait.h>
+#include <aspect/adiabatic_conditions/interface.h>
+
+
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    template <int dim>
+    void
+    ModifiedTait<dim>::
+    evaluate(const MaterialModelInputs<dim> &in,
+             MaterialModelOutputs<dim> &out) const
+    {
+      for (unsigned int i=0; i < in.temperature.size(); ++i)
+        {
+          const double temperature = in.temperature[i];
+          const double pressure = in.pressure[i];
+
+          AssertThrow(temperature > 100., ExcMessage("The current temperature is <100 K. "
+                                                     "The modified Tait equations of state is "
+                                                     "not designed for such temperatures. "
+                                                     "If this exception is triggered at the "
+                                                     "beginning of a run, you may need to "
+                                                     "set Adiabatic surface temperature "
+                                                     "to a value > 100 in the parameter file."));
+          out.viscosities[i] = eta;
+          out.thermal_conductivities[i] = k_value;
+
+          // Einstein equations for the heat capacity at constant volume
+          const double x_einstein = einstein_temperature/temperature;
+          const double x_einstein0 = einstein_temperature/reference_temperature;
+
+
+          AssertThrow(x_einstein < 35., ExcMessage("The current temperature is less than 1/35th "
+                                                   "of the Einstein temperature. This is not "
+                                                   "expected for geological materials."));
+          AssertThrow(x_einstein0 < 35., ExcMessage("The reference temperature of the "
+                                                    "equation of state is less than 1/35th "
+                                                    "of the Einstein temperature. This is not "
+                                                    "expected for geological materials."));
+
+          // The following Einstein energies and heat capacities are divided through by 3nR.
+          // This doesn't matter, as the equation of state relies only on ratios of these quantities.
+          // At large values of x_einstein0, we can simplify the calculation of the thermal energy and Cv.
+          double E_th0, E_th, C_V0, C_V;
+          E_th0 = einstein_temperature * (0.5 + 1. / (std::exp(x_einstein0) - 1.0));
+          C_V0 = x_einstein0 * x_einstein0 * std::exp(x_einstein0) / std::pow(std::exp(x_einstein0) - 1.0, 2.0);
```

Comment:
Ah, this definitely makes the code easier, doesn't it? ;-)

Since you don't modify the variables any more, declare and initialize them in line. Then you can make them `const` as well.